Python/compile.c

```diff
@@ -3655,6 +3656,27 @@ starunpack_helper(struct compiler *c, asdl_seq *elts,
 {
     Py_ssize_t n = asdl_seq_LEN(elts);
     Py_ssize_t i, nsubitems = 0, nseen = 0;
+    if (n > 1 && are_all_items_const(elts, 0, n)) {
```

Comment:
Is `n > 1` really good threshold?  How about `n > 2`?

```
$ python3 -m timeit -s "a=frozenset((1,2))" -- "{*a}"
5000000 loops, best of 5: 63.4 nsec per loop
$ python3 -m timeit "{1,2}"
5000000 loops, best of 5: 61.5 nsec per loop

$ python3 -m timeit "[*(1,2)]"
5000000 loops, best of 5: 46.5 nsec per loop
$ python3 -m timeit "[1,2]"
5000000 loops, best of 5: 46.6 nsec per loop
```

Reply:
I only chose `n > 1` because it *always* emits fewer opcodes than before the change. If you think `n > 2` is a better heuristic, I'm fine with that.

Reply:
I only think your first example is slower because it pays the price of a `LOAD_NAME` instead of `LOAD_CONST`, though. It's not a direct comparison, unfortunately.

Reply:
The fewer opcodes is not always more efficient.
BUILD_LIST_UNPACK calls `PyList_New(0)` and `PyList_Extend(iterable)`.  It causes over-allocation.
On the other hand, BUILD_LIST just calls `PyList_New(n)`.

So I think we should be conservative about choosing the threshold.

Reply:
Thanks for the feedback @methane. I think this makes sense.

I've bumped the threshold to `n > 2`. Anything else?